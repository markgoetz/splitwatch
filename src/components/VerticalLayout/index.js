import { h } from 'preact';
import styles from './style.css';

export default function VerticalLayout(props) {
  let additionalClassname = '';
  
  switch (props.align) {
  case 'left':
    additionalClassname = styles.left;
    break;
  case 'right':
    additionalClassname = styles.right;
    break;
  case 'center':
    additionalClassname = styles.center;
  }

  const className = styles.verticalLayout + ' ' + additionalClassname;

  return (<div class={className}>
    {props.children}
  </div>);
}