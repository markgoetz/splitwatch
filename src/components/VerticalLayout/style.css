.verticalLayout {
    display: flex;
    flex-direction: column;
    align-items: center;
}