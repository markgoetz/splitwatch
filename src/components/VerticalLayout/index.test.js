import { h } from 'preact';
import VerticalLayout from './index';
import { shallow } from 'preact-render-spy';

describe('Vertical Layout', () => {

  test('Vertical layout renders correctly', () => {
    const context = shallow(<VerticalLayout>
      <div>Test element</div>
      <button>Test button</button>
      <div>Extra test element</div>
    </VerticalLayout>);
    expect(context).toMatchSnapshot();
  });

  test('Vertical layout renders correctly when centered', () => {
    const context = shallow(<VerticalLayout centered>
      <div>Test element</div>
      <button>Test button</button>
      <div>Extra test element</div>
    </VerticalLayout>);
    expect(context).toMatchSnapshot();
  });
});