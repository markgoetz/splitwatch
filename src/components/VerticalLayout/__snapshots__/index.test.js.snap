// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vertical Layout Vertical layout renders correctly 1`] = `
preact-render-spy (1 nodes)
-------
<div>
  <div>Test element</div>
  <button>Test button</button>
  <div>Extra test element</div>
</div>

`;
